src/plugins/intel_gpu/src/runtime/engine.cpp

```diff
@@ -254,7 +254,13 @@ std::shared_ptr<cldnn::engine> engine::create(engine_types engine_type,
                                               const engine_configuration& configuration,
                                               const InferenceEngine::ITaskExecutor::Ptr task_executor) {
     device_query query(engine_type, runtime_type);
-    device::ptr default_device = query.get_available_devices().begin()->second;
+    GPU_DEBUG_GET_INSTANCE(debug_config);
+    device::ptr default_device = next(query.get_available_devices().begin(), debug_config->test_device_index)->second;
```

Comment:
access debug_config will cause compilation failure when GPU_DEBUG_CAPS is not enabled.

Reply:
Added #if and #ifdef to handle DEBUG_CAPS=OFF case